hub/saml-soap-proxy/src/main/java/uk/gov/ida/hub/samlsoapproxy/healthcheck/MatchingServiceHealthChecker.java

```diff
@@ -128,50 +153,52 @@ private MatchingServiceHealthCheckResult logAndCreateUnhealthyResponse(MatchingS
         return unhealthy(generateHealthCheckDescription(
                 message,
                 configEntity.getUri(),
-                Optional.<String>absent(), configEntity.isOnboarding()));
+                UNDEFINED_VERSION,
+                configEntity.isOnboarding()
+        ));
     }
 
     private MatchingServiceHealthCheckDetails generateHealthCheckDescription(
             final String message,
             final URI matchingServiceUri,
-            final Optional<String> version,
+            final String versionNumber,
             final boolean isOnboarding) {
 
-        String versionNumber = version.isPresent() ? version.get() : UNDEFINED_VERSION;
         boolean isSupported = isMsaVersionSupported(versionNumber);
         return new MatchingServiceHealthCheckDetails(matchingServiceUri, message, versionNumber,
                 isSupported, isOnboarding);
     }
 
     private boolean isMsaVersionSupported(final String versionNumber) {
-        return supportedMsaVersionsRepository.getSupportedVersions().contains(versionNumber);
+        return versionNumber != null && supportedMsaVersionsRepository.getSupportedVersions().contains(versionNumber);
```

Comment:
will versionNumber ever be null now that it's been defaulted to UNDEFINED_VERSION?